waiting_room.go

```diff
@@ -532,3 +550,56 @@ func (api *API) DeleteWaitingRoomRule(ctx context.Context, rc *ResourceContainer
 
 	return r.Result, nil
 }
+
+// WaitingRoomSettings fetches the Waiting Room zone-level settings for a zone.
+//
+// API reference: https://api.cloudflare.com/#waiting-room-get-zone-settings
+func (api *API) WaitingRoomSettings(ctx context.Context, zoneID string) (WaitingRoomSettings, error) {
+	uri := fmt.Sprintf("/zones/%s/waiting_rooms/settings", zoneID)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return WaitingRoomSettings{}, err
+	}
+	var r WaitingRoomSettingsResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return WaitingRoomSettings{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return r.Result, nil
+}
+
+// ChangeWaitingRoomSettings lets you change individual zone-level Waiting Room settings. This is
+// in contrast to UpdateWaitingRoomSettings which replaces all settings.
+//
+// API reference: https://api.cloudflare.com/#waiting-room-patch-zone-settings
+func (api *API) ChangeWaitingRoomSettings(ctx context.Context, zoneID string, settings WaitingRoomSettingsPatch) (WaitingRoomSettings, error) {
```

Comment:
this is an interesting convention. in the past, we've just supplied one method that is capable of individually updating in place for 1 or more values. i don't have a better practice to recommend at the moment so if you want to keep both, i'm 👍

Reply:
👍🏻